### PR TITLE
new trace: precision

### PR DESCRIPTION
**Pull request by geez0219**
1. complete the new trace: precision

### CODE REVIEW

File: fastestimator/trace/metric/precision.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        sample_weight (1d array-like, optional): Sample weight. If None, it will not apply sample weight. For more
+            details please refer to sklearn.metrics.precision_score. Defaults to None.
+        mode (str, optional): Restrict the trace to run only on given modes {'train', 'eval', 'test'}. None will always
+                    execute. Defaults to 'eval'.
+        output_name (str, optional): Name of the key to store to the state. Defaults to "precision".
+    """
+    def __init__(self,
+                 true_key: str,
+                 pred_key: str,
+                 labels=None,
```
Comment (TortoiseHam) [inline fastestimator/trace/metric/precision.py]: can you include typing for all args?

File: fastestimator/trace/metric/precision.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+from fastestimator.backend.to_number import to_number
+from fastestimator.trace.trace import Trace
+from fastestimator.util import Data
+
+
+class Precision(Trace):
+    """Computes precision for classification task and report it back to logger.
+
+    Args:
+        true_key (str): Name of the keys in the ground truth label in data pipeline.
```
Comment (TortoiseHam) [inline fastestimator/trace/metric/precision.py]: we're going to type hint in code instead of doc string now

File: fastestimator/trace/metric/precision.py (hunk 1 above)
Comment (geez0219) [reply fastestimator/trace/metric/precision.py]: finish adding typing

File: fastestimator/trace/metric/precision.py (hunk 2 above)
Comment (geez0219) [reply fastestimator/trace/metric/precision.py]: removed arguments type in the docstring
